### PR TITLE
BackOffice - Addresses page - Fix error after saving new address 

### LINKED ISSUE #20095: [BO Addresses page] - Error after saving new address
#### Describe the bug

An error occurs when trying to add a new address in BO

#### Expected behavior

Address saved, no errors shown

#### Steps to Reproduce

1. Go to 'BO -> Addresses page'
2. Click on 'Add new address'
3. Fill the inputs and click on save

image

#### Additional information

* PrestaShop version: branch develop
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by PululuK**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | An error occurs when trying to add a new address in BO. This bug was introduced here https://github.com/PrestaShop/PrestaShop/pull/20028
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #20095.
| How to test?  | See #20095 by @boubkerbribri .

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20149)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php [hunk 1]
```diff
@@ -292,16 +292,20 @@ public function createAction(Request $request): Response
                     );
                 }
 
-                return $this->redirectToRoute('admin_customers_view', [
-                    'customerId' => $customerId,
-                ]);
+                if ($customerId) {
```
Comment (matks) [inline src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php:295]: How is that possible that `$customerId` is null ? 🤔  I dont think it's possible to create adresses into the BO without linking them to a customer, which would consequently assign a value to `$customerId` ?

Maybe the solution here is not to handle the "what if `$customerId` is null" but rather to fetch the `$customerId` where it is.

Something like
```
$handlerResult = $addressFormHandler->handle($addressForm);

if ($handlerResult->isSubmitted() && $handlerResult->isValid()) {
    $customerId = $addressForm->getData()['customer_id'];
```

Comment (PululuK) [reply src/PrestaShopBundle/Controller/Admin/Sell/Address/AddressController.php:295]: Hi @matks 

I think :

- If you come from the customer controller, you must be redirected to the customer page after the address creation. 
- If you have created the address from address controller you must be redirected to the address page.

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @PululuK,

The exception is not displayed anymore, but I found an issue when I try to create an address with an invalid field, for example, The field Address is too long > 128 chars => redirection to the Address list page with an error.
I attached a screen record
https://drive.google.com/file/d/1zY3hXDxebLkjjx1aTaFQncB_R0i_Dlso/view?usp=sharing
With PS1.7.6.7, I have this alert
image

I created an issue for the 177x branch.
Thanks to check it.

**Comment by PululuK** [comment]
Hi @khouloudbelguith 

Coule you check if you had the same behavior before my PR ? On 1.7.7.x for exemple ? If this is the case you must open another issue, because this is not within the scope of my PR. Please check that this behavior was introduced from my PR.

Thanks

**Comment by khouloudbelguith** [comment]
@PululuK, in the develop branch without this PR, the creation of the address is blocked.
I checked the 1.7.7.x branch, I created a new issue: https://github.com/PrestaShop/PrestaShop/issues/20170

Thanks!

**Comment by khouloudbelguith** [comment]
Hi @PululuK,

This PR is ok :heavy_check_mark: 
I created the issue as it is reproduced in the 1.7.7.x branch: https://github.com/PrestaShop/PrestaShop/issues/20170

Thanks!

**Comment by matks** [comment]
Thank you @PululuK

**Pull request merged**